modules/swagger-codegen/src/main/resources/erlang-client/client_api_procession.mustache

```diff
@@ -0,0 +1,136 @@
+-module({{packageName}}_client_api_procession).
+
+-type operation_id() :: atom().
+
+-record(state, {
+    operation_id :: operation_id(),
+    validator_state :: jesse_state:state()
+}).
+
+-type state() :: state().
+
+-export([process_request/4]).
+
+-spec process_request(Method::atom(), OperationID::atom(), Url::string(), Params::map()) ->
+    {ok, integer(), list(), {ok, binary()}}.
+process_request(Method, OperationID, Url, Params) ->
+    State = #state{
+        operation_id = OperationID,
+        validator_state = prepare_validator()
+    },
+
+    {PreparedUrlParams, PreparedHeaderParams, PreparedBodyParams, PreparedQueryParams} =
+        handle_request_json(Params, State),
+
+    #{} = PreparedQueryParams,
+    PreparedUrl = prepare_url(Url, PreparedUrlParams, PreparedQueryParams),
+    PreparedHeaders = maps:to_list(PreparedHeaderParams),
+    PreparedBody = jsx:encode(PreparedBodyParams),
+
+    hackney:request(Method, PreparedUrl, PreparedHeaders, PreparedBody, [with_body]).
+
+-spec handle_request_json(term(), state()) -> {ok, map(), map(), map(), map()} | {error, term()}.
+handle_request_json(
+    RequestParams,
+    _State = #state{
+        operation_id = OperationID,
+        validator_state = ValidatorState
+    }
+) ->
+    case populate_request(OperationID, RequestParams, ValidatorState) of
+        {ok, Params} ->
+            {maps:get(binding, Params), maps:get(header, Params), maps:get(body, Params), maps:get(qs_val, Params)};
+        {error, Reason, _} ->
+            {error, Reason}
+    end.
+
+-spec populate_request(
+    OperationID :: term(),
+    RequestParams :: term(),
+    ValidatorState :: jesse_state:state()
+) ->
+    {ok, Params :: map()} |
```

Comment:
It looks like this `Params` is more then just a map, but a map with a specific set of keys (e.g. `qs_val`). How about describing it somewhere?